### PR TITLE
fix #123436: continues barlines

### PR DESCRIPTION
**Pull request by MichaelFroelich**
fix #123436: continues barlines by setting the default barline to a pre-existing barline as early as possible

### CODE REVIEW

File: libmscore/measure.cpp [hunk 1]
```diff
@@ -3418,9 +3418,17 @@ qreal Measure::createEndBarLines(bool isLastMeasureInSystem)
             seg->setEnabled(false);
             }
       else {
-            BarLineType t = nm ? BarLineType::NORMAL : BarLineType::END;
+            BarLineType t = BarLineType::NORMAL;
             if (!seg)
                   seg = getSegmentR(SegmentType::EndBarLine, ticks());
+            else {
+                  //A segment exists, therefore we should continue this barline type
+                  BarLine* bl = toBarLine(seg->element(0));
```
Comment (MarcSabatella) [inline libmscore/measure.cpp]: I guess a possible complaint is, why single out element(0)?  It *could* be the case that the top staff has a double bar the next does not, etc.  If we're to inherit an existing barline style, maybe better to pick on the staff just above?

File: libmscore/measure.cpp [hunk 2]
```diff
@@ -3418,9 +3418,17 @@ qreal Measure::createEndBarLines(bool isLastMeasureInSystem)
...
+                  if (bl && bl->enabled())
```
Comment (MarcSabatella) [inline libmscore/measure.cpp]: I think "enabled" is meaningful for the segment, not individual elements, unless barlines are different from other elements in this respect?

**Comment by MarcSabatella** [comment]
Did you test this?  Does it work for you?  It doesn't for me.  But it does seem to be on the right track.  I am taking this as a starting point and looking into this further.

**Comment by MarcSabatella** [comment]
I'm happy for you to keep working on this.  As discussed in chat, I actually don't think this approach can work for several reasons, the main one being, createBarLines() has no way of knowing which staves are new and which are not, so it's in dangerous of clobbering existing barlines (like, if you Ctrl+drag a double bar to the top staff).  Also, this won't catch barlines at the start of systems.  So at the least you'd need similar code in the function that generates those.  Better, I think, to just do all of this in Measure::cmdAddStaves(), the same way we currently fill the new measures with rests and also set up their time signatures.

The comments above on the specifics are valid too - relying on the top staff as the source of barlines means adding a new staff to top of score won't work correctly, and bl->enabled() always returns false so nothing actually happens.  It's the *segment* you might want to check for enabled, but actually, maybe you don't, because there's no reason not to set up barlines in disabled segments.  They are disabled so they can get reinstated later, and they might as well get reinstated with consistent barlines.

You will, however, want to skip repeat barlines, since those get generated automatically already.

**Comment by MarcSabatella** [comment]
I ended up including code to fix this as part of #4843, see in particular https://github.com/musescore/MuseScore/pull/4843/commits/f9780207e15db8c7a10dec8a9f699d09f92f9c9b.  I think that is the right way to fix this.  But I'm happy to revert that commit if a preferable approach is implemented here.

**Comment by MichaelFroelich** [comment]
I think it's better to decline this PR instead. Besides, I trust your fix more than mine.

**Pull request closed without merging**